### PR TITLE
Added more file type definitions. Added option to create base directo…

### PR DESCRIPTION
**Pull request by BigETI**
This pull request adds more file type definitions, an option to create base directories in `CreateDir` and the definition for an invalid directory enumerator ID.

This pull request also removes an unneeded extra abstraction layer for `std::filesystem`.

### CODE REVIEW

File: src/EFileType.hpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/// @brief File system namespace
```
Comment (Southclaws) [inline src/EFileType.hpp:1]: image

File: src/EFileType.hpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+namespace FileSystem
+{
+    /// @brief File type enumerator
```
Comment (Southclaws) [inline src/EFileType.hpp:4]: image

File: src/EFileType.hpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    enum class EFileType
+    {
+        /// @brief No file type
```
Comment (Southclaws) [inline src/EFileType.hpp:7]: image

File: src/EFileType.hpp [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        /// @brief Is character file type
+        Character,
+
+        /// @brief Is "first in, first out" file type
+        FirstInFirstOut,
+
+        /// @brief Socket file type
+        Socket,
+
+        /// @brief Unknown file type
```
Comment (Southclaws) [inline src/EFileType.hpp:34]: image

File: src/EFileType.hpp [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        /// @brief Is block file type
+        Block,
+
+        /// @brief Is character file type
+        Character,
+
+        /// @brief Is "first in, first out" file type
+        FirstInFirstOut,
+
+        /// @brief Socket file type
```
Comment (Southclaws) [inline src/EFileType.hpp:31]: image

File: src/EFileType.hpp [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        /// @brief Is symbolic link file type
+        SymbolicLink,
+
+        /// @brief Is block file type
+        Block,
+
+        /// @brief Is character file type
+        Character,
+
+        /// @brief Is "first in, first out" file type
```
Comment (Southclaws) [inline src/EFileType.hpp:28]: image

File: src/EFileType.hpp [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        /// @brief Is directory file type
+        Directory,
+
+        /// @brief Is symbolic link file type
+        SymbolicLink,
+
+        /// @brief Is block file type
+        Block,
+
+        /// @brief Is character file type
```
Comment (Southclaws) [inline src/EFileType.hpp:25]: image

File: src/EFileType.hpp [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        /// @brief Is regular file type
+        Regular,
+
+        /// @brief Is directory file type
+        Directory,
+
+        /// @brief Is symbolic link file type
+        SymbolicLink,
+
+        /// @brief Is block file type
```
Comment (Southclaws) [inline src/EFileType.hpp:22]: image

File: src/EFileType.hpp [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+        /// @brief File not found
+        NotFound,
+
+        /// @brief Is regular file type
+        Regular,
+
+        /// @brief Is directory file type
+        Directory,
+
+        /// @brief Is symbolic link file type
```
Comment (Southclaws) [inline src/EFileType.hpp:19]: image

File: src/EFileType.hpp [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+        /// @brief No file type
+        None,
+
+        /// @brief File not found
+        NotFound,
+
+        /// @brief Is regular file type
+        Regular,
+
+        /// @brief Is directory file type
```
Comment (Southclaws) [inline src/EFileType.hpp:16]: image

File: src/EFileType.hpp [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    /// @brief File type enumerator
+    enum class EFileType
+    {
+        /// @brief No file type
+        None,
+
+        /// @brief File not found
+        NotFound,
+
+        /// @brief Is regular file type
```
Comment (Southclaws) [inline src/EFileType.hpp:13]: image

File: src/EFileType.hpp [hunk 12]
```diff
@@ -0,0 +1,37 @@
+/// @brief File system namespace
+namespace FileSystem
+{
+    /// @brief File type enumerator
+    enum class EFileType
+    {
+        /// @brief No file type
+        None,
+
+        /// @brief File not found
```
Comment (Southclaws) [inline src/EFileType.hpp:10]: image

File: src/fsutil.cpp [hunk 13]
```diff
@@ -1,55 +0,0 @@
-#include <amx/amx.h>
```
Comment (Southclaws) [inline src/fsutil.cpp:1]: Renaming seems pointless, it just puts a big gap in the history.

**Review by Southclaws — COMMENTED** [review]
It's basically a full rewrite, which is fine I guess, just make sure the tests pass on both OSes.

File: src/EFileType.hpp (hunk 1 above)
Comment (ADRFranklin) [reply src/EFileType.hpp:1]: LMAO, best comment

**Pull request merged**